### PR TITLE
fix compile error + temporary fix width/height setting

### PR DESCRIPTION
**Pull request by anneb**
This PR fixes compile errors for undefined variables.
It also temporarily fixes picture width/height setting to preview.width/preview.height as all other values result in garbled output images.

### CODE REVIEW

**Comment by westonganger** [comment]
I see you have commented out some code there, are you still working on the correct implementation?

File: src/android/CameraActivity.java [hunk 1]
```diff
@@ -348,15 +348,45 @@ public void onPreviewFrame(final byte[] data, final Camera camera) {
...
+                    Log.d(TAG, "bestMatch: " + bestMatch + ", width: " + w + ", height: " + h);
+                  }
+                }
+                Log.d(TAG, "Calculated picture width: " + w + ", height: " + h);
               }
-
+              */
+              Camera.Parameters params = camera.getParameters();
+              w = params.getPreviewSize().width;
+              h = params.getPreviewSize().height;
```
Comment (westonganger) [inline src/android/CameraActivity.java]: Shouldn't these 3 lines be inside the if statement?

```
if(w == 0 || h == 0){
  /* 
    commented code
  */

  Camera.Parameters params = camera.getParameters();
  w = params.getPreviewSize().width;
  h = params.getPreviewSize().height;
}
```

**Comment by anneb** [comment]
These lines are outside the if statement, because any other value of w and h will produce garbled images.

The picture is taken from the preview, so the picture width and height must be equal to the width and height.

The commented code was assuming that the picture frame was coming from the camera, not from the preview.

**Comment by westonganger** [comment]
What are the preview sizes compared to what your put into startCamera?

Don't we want the photo to come from the Picture frame not the preview? Taking a picture of the preview is why we are getting low res images in #182 isn't it?

**Comment by anneb** [comment]
The preview size is calculated in Camera.java private getOptimalPreviewSize.
Here it looks for the highest preview resolution available below the optionally given width and height. The preview (java variable) resolution can be much better than the actual on-screen preview.

**Comment by westonganger** [comment]
Okay I am going to merge this but another fix that uses the passed in width/height is still needed.

**Pull request merged**